server/controllers/article.js

```diff
@@ -41,3 +39,22 @@ export const getArticles = async (req, res) => {
     displayError(err, res, 500);
   }
 };
+
+export const getAnArticle = async (req, res) => {
+  try {
+    const { articleId } = req.params;
+
+    const article = await Article.findOne({
+      where: { id: articleId }
+    });
+
+    if (article === null) {
+      handleFailedResponse(null, 'No article with this id found', res, 404);
+    } else {
+      handleSuccessResponse(article, 'Article retrieved successfully', res, 200);
+    }
+  } catch (error) {
+    const err = new Error('Server error');
+    displayError(err, res, 500);
+  }
+};
```

Comment:
another observation, this is a controller class and it's containing too many validations.
You might want to put that in a middleware.
...In your middleware, you should also consider validating user input

Reply:
These are responses that I think are meant to be handled in the controller and not validations.